packages/react-reconciler/src/ReactFiberCallUserSpace.js

```diff
@@ -177,11 +179,22 @@ export const callComponentWillUnmountInDEV: (
 
 const callCreate = {
   'react-stack-bottom-frame': function (effect: Effect): (() => void) | void {
-    const create = effect.create;
-    const inst = effect.inst;
-    const destroy = create();
-    inst.destroy = destroy;
-    return destroy;
+    if (enableUseResourceEffectHook && typeof effect.destroy === 'function') {
+      const inst = effect.inst;
+      const destroy_ = effect.destroy;
+      const destroy = () => {
+        destroy_(effect.resource);
+        effect.resource = null;
+      };
+      inst.destroy = destroy;
+      return destroy;
+    } else if (effect.kind === SimpleEffectKind) {
```

Comment:
The feature flagging here is weird to me, why is this condition necessary if `enableUseResourceEffectHook` is false? When the flag gets stripped, you should get the previous behavior without the flag, but I think this will include this new check.